packages/ui/internal/validators/tests/validators-specs.js

```diff
@@ -0,0 +1,124 @@
+import {warn, validateRange, validateStepped, validateRangeOnce, validateSteppedOnce} from '../validators';
+
+describe('validators', () => {
+	test('should throw a console warning', () => {
+		const consoleSpy = jest.spyOn(console, 'warn').mockImplementation(() => {});
+		warn('Warning!');
+
+		expect(consoleSpy).toHaveBeenCalled();
+	});
+
+	test('should throw a console warning when \'value\' is less than \'min\'', () => {
+		const consoleSpy = jest.spyOn(console, 'warn').mockImplementation(() => {});
+		validateRange(10, 11, 20, 'component', 'value', 'min', 'max');
+
+		expect(consoleSpy).toHaveBeenCalled();
+	});
+
+	test('should throw a console warning when \'value\' is greater than \'max\'', () => {
+		const consoleSpy = jest.spyOn(console, 'warn').mockImplementation(() => {});
+		validateRange(21, 1, 20, 'component', 'value', 'min', 'max');
+
+		expect(consoleSpy).toHaveBeenCalled();
+	});
+
+	test('should throw a console warning when \'min\' is greater than \'max\'', () => {
+		const consoleSpy = jest.spyOn(console, 'warn').mockImplementation(() => {});
+		validateRange(10, 21, 20, 'component', 'value', 'min', 'max');
+
+		expect(consoleSpy).toHaveBeenCalled();
+	});
+
+	test('should throw a console warning when \'value\' is evenly divisible by \'step\'', () => {
+		const consoleSpy = jest.spyOn(console, 'warn').mockImplementation(() => {});
+		validateStepped(10, 1, 2, 'component', 'value', 'step');
+
+		expect(consoleSpy).toHaveBeenCalled();
+	});
+
+	describe('validateRangeOnce', () => {
+		let thingSpy;
+
+		beforeEach(() => {
+			thingSpy = jest.fn().mockReturnValue("ANY_THING_SPY_VALUE");
+		});
+
+		test('should throw a console warning when \'value\' is lower than \'min\'', () => {
+			const consoleSpy = jest.spyOn(console, 'warn').mockImplementation(() => {});
+			const validateFn = validateRangeOnce(thingSpy, {
+				component: "ANY_COMPONENT"
+			});
+
+			const normalizedValues = validateFn({
+				value: 1,
+				min: 2,
+				max: 10
+			});
+
+			expect(consoleSpy).toHaveBeenCalled();
+			expect(thingSpy).toHaveBeenCalled();
+			expect(consoleSpy.mock.calls).toMatchSnapshot(`
```

Comment:
Is this snapshot testing necessary? I think we could use https://jestjs.io/docs/expect#expectarraycontainingarray or other methods.
Otherwise, you need to commit the snapshot file to git as https://jestjs.io/docs/snapshot-testing informed. I think snapshot testing is checking rendered HTML or so.